src/main/java/org/elasticsearch/rest/action/search/RestClearScrollAction.java

```diff
@@ -47,16 +58,46 @@ public RestClearScrollAction(Settings settings, RestController controller, Clien
 
     @Override
     public void handleRequest(final RestRequest request, final RestChannel channel, final Client client) {
-        String scrollIds = request.param("scroll_id");
-        if (scrollIds == null) {
-            scrollIds = RestActions.getRestContent(request).toUtf8();
-        }
-
         ClearScrollRequest clearRequest = new ClearScrollRequest();
-        clearRequest.setScrollIds(Arrays.asList(splitScrollIds(scrollIds)));
+
+        parseContent(clearRequest, request.param("scroll_id"), RestActions.getRestContent(request));
         client.clearScroll(clearRequest, new RestStatusToXContentListener<ClearScrollResponse>(channel));
     }
 
+    public static void parseContent(ClearScrollRequest request, String scrollIds, BytesReference content) {
```

Comment:
Also move the parsing logic to the transport action layer? Because there is just one option on the request there should just be a scroll id setter that transforms the ids into json format.